src/serac/physics/tests/test_kinematic_objective.cpp

```diff
@@ -0,0 +1,176 @@
+
+    double mass = mass_objective.evaluate(time, dt, objective_states);
+
+    serac::tensor<double, dim> initial_cg;
+
+    for (int i = 0; i < dim; ++i) {
+      auto cg_objective = std::make_shared<ObjectiveT>("translation" + std::to_string(i), mesh,
+                                                       all_states[SHAPE_DISP]->space(), param_space_ptrs);
+      cg_objective->addBodyIntegral(
+          serac::DependsOn<0, 1>{}, mesh->entireBodyName(),
+          [i](double
+              /*time*/,
+              auto X, auto U,
+              auto RHO) { return (get<serac::VALUE>(X)[i] + get<serac::VALUE>(U)[i]) * get<serac::VALUE>(RHO); });
+      initial_cg[i] = cg_objective->evaluate(time, dt, objective_states) / mass;
+      constraint_evaluators.push_back(cg_objective);
+    }
+
+    for (int i = 0; i < dim; ++i) {
+      auto center_rotation_objective = std::make_shared<ObjectiveT>("rotation" + std::to_string(i), mesh,
+                                                                    all_states[SHAPE_DISP]->space(), param_space_ptrs);
+      center_rotation_objective->addBodyIntegral(serac::DependsOn<0, 1>{}, mesh->entireBodyName(),
+                                                 [i, initial_cg](double /*time*/, auto X, auto U, auto RHO) {
+                                                   auto u = get<serac::VALUE>(U);
+                                                   auto x = get<serac::VALUE>(X) + u;
+                                                   auto dx = x - initial_cg;
+                                                   auto x_cross_u = serac::cross(dx, u);
+                                                   return x_cross_u[i] * get<serac::VALUE>(RHO);
+                                                 });
+      constraint_evaluators.push_back(center_rotation_objective);
+    }
+
+    return constraint_evaluators;
+  }
+
+  void SetUp()
+  {
+    MPI_Barrier(MPI_COMM_WORLD);
+    serac::StateManager::initialize(datastore, "solid_dynamics");
+
+    double xlength = 0.5;
+    double ylength = 0.7;
+    double zlength = 0.3;
+    mesh = std::make_shared<serac::Mesh>(mfem::Mesh::MakeCartesian3D(6, 4, 4, element_shape, xlength, ylength, zlength),
+                                         "this_mesh_name", 0, 0);
+
+    serac::FiniteElementState disp = serac::StateManager::newState(VectorSpace{}, "displacement", mesh->tag());
+    serac::FiniteElementState velo = serac::StateManager::newState(VectorSpace{}, "velocity", mesh->tag());
+    serac::FiniteElementState accel = serac::StateManager::newState(VectorSpace{}, "acceleration", mesh->tag());
+    serac::FiniteElementState shape_disp = serac::StateManager::newState(VectorSpace{}, "shape_disp", mesh->tag());
+    serac::FiniteElementState density = serac::StateManager::newState(DensitySpace{}, "density", mesh->tag());
+
+    states = {disp, velo, accel, shape_disp};
```

Comment:
Should this be consistent with ordering of the `FIELD` enum and be `states = {shape_disp, disp, velo, accel};`?